Sources/TuistCache/Utilities/CacheXCFrameworkBuilder.swift

```
@@ -15,14 +15,20 @@ public final class CacheXCFrameworkBuilder: CacheArtifactBuilding {
 
     /// Initializes the builder.
     /// - Parameter xcodeBuildController: Xcode build controller instance to run xcodebuild commands.
-    public init(xcodeBuildController: XcodeBuildControlling) {
+    /// - Parameter cacheOutputType: Output type of xcframework (device and/or simulator)
+    public init(
+        xcodeBuildController: XcodeBuildControlling,
+        cacheOutputType: CacheOutputType
+    ) {
+        precondition(cacheOutputType.isXCFramework, "cacheOutputType of CacheXCFrameworkBuilder must be a xcframework type")
```

Should we throw an error here instead?